services/github/github-go-mod.tester.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const { isVPlusDottedVersionAtLeastOne } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Go version')
+  .get('/gohugoio/hugo.json')
+  .expectBadge({
+    label: 'Go',
+    message: isVPlusDottedVersionAtLeastOne,
+  })
+
+t.create('Go version (from branch)')
+  .get('/gohugoio/hugo/master.json')
+  .expectBadge({
+    label: 'Go@master',
+    message: isVPlusDottedVersionAtLeastOne,
+  })
+
+t.create('Go version (mongorepo)')
+  .get(`/golang/go.json?filename=${encodeURIComponent('src/go.mod')}`)
+  .expectBadge({
+    label: 'Go',
+    message: isVPlusDottedVersionAtLeastOne,
+  })
+
+t.create('Go version (repo not found)')
+  .get('/badges/not-existing-repo.json')
+  .expectBadge({
+    label: 'Go',
+    message: 'repo not found, branch not found, or go.mod missing',
+  })
+
+t.create('Go version (missing Go version in go.mod)')
```

Comment:
Could this be changed to a live test, now that there's a way to specify an alternate filename?

Reply:
It would require a `go.mod` file or in fact any kind of file that exists and and does not contain a line similar to `go 1.14`. Personally, in this case, I would stick to a mock server as it allows control over the returned content. A live test gives you certainty the badge communicates correctly with the server. This test verifies how badge behaves when a file doesn't contain expected line. In my opinion a "traditional" unit test suits here better.

If you guys feel it's better having a live test I can make it by pointing to `shields`'s repo `package.json` for instance.

Reply:
I just forked a repo and removed the go version, feel free to use this as the target for a live test

https://github.com/calebcartwright/ci-detective/blob/master/go.mod

Reply:
Thanks. I've changed the test to live one.